### PR TITLE
NO-TICKET - Add some more info about swap validation

### PR DESCRIPTION
**Pull request by gslevinkov**

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt [hunk 1]
```diff
@@ -139,8 +140,10 @@ class RpcInterceptor(
 
         val errorMessage = serverError.error.data?.getErrorLog() ?: serverError.error.message
 
-        val errorType = serverError.error.data?.rpcErrorDetails
-        val domainErrorType = runCatching {
+        val errorType: JsonElement? = serverError.error.data
+            ?.rpcErrorDetails
+            ?.takeIf { !it.isJsonNull }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt:145]: error: null is parsed as JsonElement.Null and not JsonElement?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/finance_block/FinanceBlockDelegates.kt [hunk 2]
```diff
@@ -12,7 +12,6 @@ private typealias OnBindListener = (view: UiKitMainCellView, item: MainCellModel
 fun mainCellDelegate(
     inflateListener: ((financeBlock: UiKitMainCellView) -> Unit)? = null,
     onBindListener: ((view: UiKitMainCellView, item: MainCellModel) -> Unit)? = null,
-    onItemClicked: (item: MainCellModel) -> Unit = {}
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/finance_block/FinanceBlockDelegates.kt:15]: didn't work

File: app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt:145]: json null value is the JsonElement (JsonNull)

File: app/src/main/java/org/p2p/wallet/jupiter/repository/routes/JupiterSwapRouteValidator.kt [hunk 3]
```diff
@@ -66,11 +77,12 @@ class JupiterSwapRouteValidator(
         return SwapRouteValidation(
             route = route,
             ordinal = ordinal,
-            isRouteValid = isRouteValid
+            isRouteValid = isRouteValid.isSimulationSuccess,
+            errorIfNotValid = isRouteValid.errorIfSimulationFailed
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/jupiter/repository/routes/JupiterSwapRouteValidator.kt:81]: maybe just `error` instead of `errorIfNotValid`?

**Review by eduardmaximovich — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**